OpenFeature/src/test/java/dev/openfeature/sdk/TestFeatureProvider.kt

```diff
@@ -0,0 +1,79 @@
+package dev.openfeature.sdk
+
+import dev.openfeature.sdk.events.EventHandler
+import dev.openfeature.sdk.events.OpenFeatureEvents
+import kotlinx.coroutines.CoroutineDispatcher
+
+class TestFeatureProvider(
```

Comment:
Can this be the in-memory provider? We started adding those in to other SDKs. It can be used in tests or for basic use cases.

Reply:
In most SDKs there's a testing provider (or a few) that have more testing features related to testing the SDK itself (such as throwing on init to test init logic, etc). That seems more like what this is.

The [in-memory provider](https://openfeature.dev/specification/appendix-a#in-memory-provider) is mostly for outsiders to test things. I think that's a separate issue.